src/index.ts

```diff
@@ -149,29 +209,58 @@ function findTargetPath(pathMatches) {
   return operation;
 }
 
-class Oas {
-  constructor(oas, user) {
-    this.api = oas || {};
+export default class Oas {
+  /**
+   * OpenAPI API Definition that this instance should use.
+   */
+  api: RMOAS.OASDocument;
+
+  /**
+   * The current user that we should use when pulling auth tokens from security schemes.
+   */
+  user: RMOAS.User;
+
+  /**
+   * Internal storage array that the library utilizes to keep track of the times the {@see Oas.dereference} has been
+   * called so that if you initiate multiple promises they'll all end up returning the same data set once the initial
+   * dereference call completed.
+   *
+   * @todo These are always `Promise.resolve` and `Promise.reject`. Make these types more specific than `any`.
+   */
+  protected promises: Array<{
+    resolve: any;
+    reject: any;
+  }>;
+
+  /**
+   * Internal storage array that the library utilizes to keep track of its `dereferencing` state so it doesn't initiate
+   * multiple dereferencing processes.
+   */
+  protected dereferencing: {
+    processing: boolean;
+    complete: boolean;
+  };
+
+  constructor(oas: RMOAS.OASDocument, user?: RMOAS.User) {
+    this.api = oas || ({} as RMOAS.OASDocument);
```

Comment:
Yeah this throws because things that are required for a valid OAS aren't present in `{}`:

```
Type 'OASDocument | {}' is not assignable to type 'OASDocument'.
  Type '{}' is not assignable to type 'OASDocument'.
    Type '{}' is not assignable to type 'Omit<Omit<Document<{}>, "paths" | "components">, "servers" | "paths" | "components" | "info" | "webhooks" | "jsonSchemaDialect"> & { info: InfoObject; jsonSchemaDialect?: string; servers?: ServerObject[]; } & Pick<...> & Omit<...>'.
      Property 'openapi' is missing in type '{}' but required in type 'Omit<Omit<Document<{}>, "paths" | "components">, "servers" | "paths" | "components" | "info" | "webhooks" | "jsonSchemaDialect">'.ts(2322)
index.d.ts(153, 9): 'openapi' is declared here.
```

Typecasting it directly to `as RMOAS.OASDocument` fixes that. I don't really want to define `this.api` as `Partial<RMOAS.OASDocument>` because that's going to make everything in the OAS that's required as not.